metricbeat/module/windows/wmi/_meta/docs.asciidoc

```diff
@@ -1,35 +1,114 @@
 Systems.
 
-This metricset leverages the link:https://github.com/microsoft/wmi[Microsoft WMI], library a
-convenient wrapper around the link:https://github.com/go-ole[GO-OLE] library which allows to
-invoke the WMI Api.
+This metricset leverages the link:https://github.com/microsoft/wmi[Microsoft WMI] library, a
+convenient wrapper around the link:https://github.com/go-ole[Go-OLE]. This allows invoking the
+link:https://learn.microsoft.com/en-us/windows/win32/wmisdk/scripting-api-for-wmi[Scripting API for WMI].
 
+[float]
 === WMI Query Language (WQL) Support
 
 This metricset supports the execution of link:https://learn.microsoft.com/en-us/windows/win32/wmisdk/wql-sql-for-wmi[WQL] queries, a SQL-like query language for retrieving information from WMI namespaces.
 
 As of now, we only support and execute queries with `SELECT`, `FROM` and `WHERE` clauses.
 
+[float]
 === Configuration
 
 [source,yaml]
 ----
 - module: windows
   metricsets: ["wmi"]
   period: 60s
-  namespace: "root\\cimv2" # Namespace
-  queries:
-  - class: Win32_OperatingSystem
-    fields:
-    - FreePhysicalMemory
-    - FreeSpaceInPaginFiles
-    - NumberOfUsers
-    # Where Clasue
-    where: ""
+  wmi:
+    # host: localhost
+    # domain: ""
+    # username: test
+    # password: foo
+    namespace: "root\\cimv2" # Default Namespace
+    warning_threshold: 60s
+    include_queries: true
+    include_null: false
+    include_empty_strings: false
+    queries:
+    - class: Win32_OperatingSystem
+      fields:
+      - FreePhysicalMemory
+      - FreeSpaceInPagingFiles
+      - NumberOfUsers
+      # Where Clasue
+      where: ""
+    - class: Win32_PowerPlan
+      fields: []
+      where: "IsActive = True"
+      namespace: "root\\cimv2\\power" # Overwrites the module namespace in this query
+----
+
+
+*`wmi.host`*:: The host for the remote WMI Connection. By default is localhost.
+
+*`wmi.domain`*:: The domain for the remote WMI Connection. By default is the empty string.
+
+*`wmi.username`*:: The username for the remote WMI Connection. By default is empty string.
+
+*`wmi.password`*:: The password for the remote WMI Connection. By default is empty string.
+
+*`wmi.warning_threshold`*:: The time threshold after which Metricbeat will stop
+waiting for the query result and return control to the main flow of the program.
+A warning is logged indicating that the query execution has exceeded the threshold.
+By default is equal to the period.
+
+*`wmi.queries`*:: The list of queries to execute. The list cannot be empty. See <<query-configuration, Query Configuration>> for the format of the queries.
+
+*`wmi.include_queries`*:: A boolean option that causes the metricset to include the
+query in the output document. False by default.
+
+*`wmi.include_null`*:: A boolean option that causes the metricset to include the
+properties that have a null value in the output document. False by default.
+
+*`wmi.include_empty_string`*:: A boolean option that causes the metricset to include
```

Comment:
```suggestion
*`wmi.include_empty_strings`*:: A boolean option that causes the metricset to include
```

Reply:
If we do that, we need to fix the go code as well. 

I think maybe we can rename it to `wmi.include_empty_string_properties` to make it more clear 🤔 (The same holds for null `wmi.include_null_properties`). WDYT?

Reply:
What do we mean by properties here ?
Are we saying fields that are null/empty strings?

Reply:
Yes sorry. If we stick with the fields like in old wmibeats then it's better wmi.include_empty_string_fields, wmi.include_null_fields. 

But probably it would make sense to use the term "Properties" instead of "fields" https://learn.microsoft.com/en-us/windows/win32/wmisdk/wmi-classes#naming-conventions-for-wmi-classes-and-properties to be more compliant with WMI terminology

Reply:
Then we can name it as wmi.include_null_properties and wmi.include_empty_string_properties ?

Reply:
I fix this in the new version. I performed a refactoring to make sure we always speak about properties instead of fields (indeed, WMI never mention fields). 

Now we have:
`wmi.include_empty_string_properties` and `wmi.include_null_properties`, which are also self-explanatory.